models/remote_auth.go

```diff
@@ -216,31 +224,57 @@ func (l *MesheryRemoteProvider) GenerateKey(jwk JWK) (*rsa.PublicKey, error) {
 func (l *MesheryRemoteProvider) VerifyToken(tokenString string) (*jwt.MapClaims, error) {
 	dtoken, err := l.DecodeTokenData(tokenString)
 	if err != nil {
-		logrus.Error("E1")
+		logrus.Error("error decoding token : %v", err.Error())
 		return nil, err
 	}
 	tokenString = dtoken.AccessToken
-	tokenUP, _, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
+	tokenUP, x, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
 	if err != nil {
-		logrus.Error("2")
+		logrus.Error("error parsing token (unverified) : %v", err.Error())
 		return nil, err
 	}
 	kid := tokenUP.Header["kid"].(string)
+
+	var jtk map[string]interface{}
+	t, _ := base64.RawStdEncoding.DecodeString(x[1])
+	json.Unmarshal(t, &jtk)
+
+	// TODO: Once hydra fixes https://github.com/ory/hydra/issues/1542
+	// we should rather configure hydra auth server to remove nbf field in the token
+	exp := int64(jtk["exp"].(float64))
+	if jwt.TimeFunc().Unix() > exp {
+		err := fmt.Errorf("Token has expired")
+		logrus.Errorf("error validating token : %v", err.Error())
+		return nil, err
+	}
+
+	// Enable only for Debugging
```

Comment:
This block can be removed, not an important block. User can write manually for debugging or use the debugger.